app/controllers/api/v1/deliveries_controller.rb

```diff
@@ -104,14 +104,22 @@ def confirm_delivery
       @delivery.gogovan_order = GogovanOrder.book_order(current_user,
         order_params) if params["gogovanOrder"]
       if @delivery && @delivery.update(get_delivery_details)
-        render json: @delivery, serializer: serializer
+        render_delivery_json
       else
-        render json: @delivery.errors.to_json, status: 422
+        render_errors
       end
     end
 
     private
 
+    def render_delivery_json
+      render json: @delivery, serializer: serializer
+    end
```

Comment:
yes @steveyken agreed.
Just added it, as it was repeated at two places, in `update` and `show` methods.
I will revert it back. Thanks for reviewing :)